tests/test_custom_models.py

```diff
@@ -868,6 +869,46 @@
         DeloraConfig,
         {"target_modules": ["lin0"], "module_dropout": 0.1},
     ),
+    ##########
+    # BD-LoRA #
+    ##########
+    (
+        "BD-LoRA A only",
+        "MLP",
+        LoraConfig,
+        {
+            "target_modules": ["lin0", "lin1"],
+            "use_bdlora": BdLoraConfig(target_modules_bd_a=["lin0"], nblocks=2, match_strict=False),
+        },
+    ),
+    (
+        "BD-LoRA B only",
+        "MLP",
+        LoraConfig,
+        {
+            "target_modules": ["lin0", "lin1"],
+            "use_bdlora": BdLoraConfig(target_modules_bd_b=["lin1"], nblocks=2, match_strict=False),
+        },
+    ),
+    (
+        "BD-LoRA both A and B",
+        "MLP",
+        LoraConfig,
+        {
+            "target_modules": ["lin0", "lin1"],
+            "use_bdlora": BdLoraConfig(target_modules_bd_a=["lin0"], target_modules_bd_b=["lin1"], nblocks=2),
+        },
+    ),
+    (
+        "BD-LoRA with modules_to_save",
+        "MLP",
+        LoraConfig,
+        {
+            "target_modules": ["lin0"],
+            "modules_to_save": ["lin1"],
+            "use_bdlora": BdLoraConfig(target_modules_bd_a=["lin0"], nblocks=2),
+        },
+    ),
```

Comment:
Can we get an example with `match_strict=False` and one vanilla LoRA layer? On the other hand, no need for the `modules_to_save` example, as I don't think there is any interaction there with BD-LoRA.